src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/course/domain/CourseService.kt

```diff
@@ -12,22 +12,22 @@ class CourseService(
   @Autowired
   val courseRepository: CourseRepository,
 ) {
-  fun allCourses(): List<CourseEntity> = courseRepository.allCourses().filterNot(CourseEntity::withdrawn)
+  fun getAllCourses(): List<CourseEntity> = courseRepository.allCourses().filterNot(CourseEntity::withdrawn)
 
-  fun course(courseId: UUID): CourseEntity? = courseRepository.course(courseId)?.takeIf { !it.withdrawn }
-  fun getCourseForOfferingId(offeringId: UUID): CourseEntity? = courseRepository.findCourseByOfferingId(offeringId)
+  fun getCourseById(courseId: UUID): CourseEntity? = courseRepository.course(courseId)?.takeIf { !it.withdrawn }
+  fun getCourseByOfferingId(offeringId: UUID): CourseEntity? = courseRepository.findCourseByOfferingId(offeringId)
 
-  fun allOfferings(): List<Offering> = courseRepository.allOfferings().filterNot(Offering::withdrawn)
+  fun getOfferingsCsv(): List<Offering> = courseRepository.allOfferings().filterNot(Offering::withdrawn)
 
-  fun offeringsForCourse(courseId: UUID): List<Offering> = courseRepository
+  fun getAllOfferingsByCourseId(courseId: UUID): List<Offering> = courseRepository
     .offeringsForCourse(courseId)
     .filterNot(Offering::withdrawn)
 
-  fun courseOffering(offeringId: UUID): Offering? = courseRepository
+  fun getOfferingById(offeringId: UUID): Offering? = courseRepository
     .courseOffering(offeringId)
     ?.takeIf { !it.withdrawn }
 
-  fun updateCourses(courseData: List<CourseUpdate>) {
+  fun uploadCoursesCsv(courseData: List<CourseUpdate>) {
```

Comment:
Do we need the `csv` in here? Is it to specify that this is only called when we upload a CSV of courses? I wonder if `update` captures the intent here slightly better, that we're using it to update courses, rather than just uploading a CSV. The only thing I can think of is `updateCoursesWithCsvData` but I don't know if that's quite right.

Reply:
Yep, again, good spot.